Sources/Tracking/CustomerIOInstance.swift

```diff
@@ -0,0 +1,63 @@
+import Common
+import Foundation
+
+public protocol CustomerIOInstance: AutoMockable {
```

Comment:
I actually prefer this protocol existing in the `CustomerIO` class file. It's a similar design pattern we do in other classes in the code base and in Android where `interface` and implementation `class` exist in the same file. 

Is there a reason why we should move it to another file? Did you encounter a swift lint error maybe?

Reply:
Hmm, definitely interested in this conversation. Without context having one file do one thing is often a good model. The other thing is if more than one thing implements the interface it makes sense. Like actual implementation and test implementation would be enough in my book. If we have a pattern we should probably stick to it, or if we are redefining it...do that.

Reply:
It was the linting error, the class is getting bigger and bypassing the number of lines approved by the linter. So this protocol seemed like the best candidate to be taken out of the class. 

I am not against either but wanted to play by the linting rules too. :)

Reply:
Because the code base currently follows a pattern of: Have 1 file that contains both the protocol and it's class that inherits the protocol, I suggest we put this protocol back into `CustomerIO.swift` file. 

I understand that this could create a lint error with the `CustomerIO` class being long. The `CustomerIO` class has lots of functions inside of it and, luckily, this is the only class at the moment that triggers this lint error. 

I am not sure on the best solution for this lint error, but I would prefer that we keep the current code base pattern consistent. I am OK with ignoring the lint error in the `CustomerIO` class at this time because I am not sure on another good solution at the moment and this is the only file at this time that breaks this lint error. Making me think we don't have a major problem.

Reply:
As long as those linting errors might not interrupt the CI workflow? 

Is there any way we can disable a particular lint error on a file? like the way, we have for Android?